### DISCUSSION TITLE
Could give examples of projects using IDOM?

### DISCUSSION THREAD

**Discussion by atilagf1** [idom-team/idom#904] [General]
I'm studying a frontend framework that I could use for my project, it's IDOM being used for some relevant projects?

**Comment by rmorshea** [comment]
The largest project I'm aware of using IDOM (soon to be [renamed ReactPy](https://github.com/idom-team/idom/discussions/882)) is [Conreq](https://github.com/Archmonger/Conreq/tree/app_store) which is being developed by @Archmonger. @numpde also seems to be a relatively active user, however I am not actually aware of what they're using IDOM for.

With all that said, it's a little hard to judge whether or not IDOM is the right choice for you without knowing more about your use case. In general, I think the nearest alternatives to IDOM are probably [Streamlit](https://streamlit.io/) and [Jupyter Widgets](https://ipywidgets.readthedocs.io/en/latest/). However IDOM has two main advantages over these solutions:

1. It's an extremely flexible framework based on software patterns that have been proven by [ReactJS](https://reactjs.org/) to scale as a project grows and ages. Ultimately this makes it easier to create more complex applications with IDOM than you might with one of its competitors.
2. Components developed for IDOM are portable to other frameworks (like [Jupyter](https://github.com/idom-team/idom-jupyter) and [Plotly Dash](https://github.com/idom-team/idom-dash)) which cannot be said of any of the alternatives.

**Reply by atilagf1** [reply]
Thank you for your answer. I'm working on projects for cryptocurrency markets, like bots and trading strategies. I would need frontended for this projects. Do you think IDOM would be a good options? I'm a backend python programmer, I was searching for some python frontend options instead of use javascript.

**Reply by rmorshea** [reply]
I think IDOM could be a great choice to start out since it makes developing applications so quick and easy. If your application grows substantially though, your probably going to need to shift away from IDOM. At a base level, when you choose IDOM, you're trading performance and scalability for development velocity. Given this potential need to transition away from IDOM as performance and scale become more important, IDOM provides a couple different ways to accomplish this. You can either develop [Custom Javascript Components](https://idom-docs.herokuapp.com/docs/guides/escape-hatches/javascript-components.html#custom-javascript-components) in order to keep more of the work in the client, or you can [embed IDOM](https://idom-docs.herokuapp.com/docs/guides/getting-started/running-idom.html#embed-in-an-existing-webpage) into a traditional application in order to limit your use of IDOM altogether.

**Reply by atilagf1** [reply]
Great answer!
